lib/handler/headers_util.c

```diff
@@ -50,7 +130,15 @@ void h2o_headers_append_command(h2o_headers_command_t **cmds, int cmd, h2o_iovec
     new_cmds = h2o_mem_alloc_shared(NULL, (cnt + 2) * sizeof(*new_cmds), NULL);
     if (*cmds != NULL)
         memcpy(new_cmds, *cmds, cnt * sizeof(*new_cmds));
-    new_cmds[cnt] = (h2o_headers_command_t){cmd, name, value, when};
+    new_cmds[cnt] = (h2o_headers_command_t){};
+    new_cmds[cnt].cmd = cmd;
+    new_cmds[cnt].when = when;
+    new_cmds[cnt].args = h2o_mem_alloc_shared(NULL, sizeof(*new_cmds->args) * num_args, NULL);
```

Comment:
To me it seems that memory allocated here is not released when `new_cmds` gets released?

I think we should:
* change `h2o_mem_alloc_shared` used here to `h2o_mem_alloc`
* create and provide a custom dispose callback when allocating `new_cmds` right above, that calls `free` for each  `cmds[x].args`.

Reply:
Good catch, fixed in 9967b65a1